code/datums/components/bayonet_charge.dm

```diff
@@ -0,0 +1,172 @@
+
+/datum/component/bayonetcharge
+	//these variables modify the stats of a bayonet charge
+	///the amount of tiles you need to walk to start charging
+	var/minactivationrundist = 5
+	///thhe amount of tiles you need to walk to charge fully
+	var/fullspeedrundist = 9
+	///speed increase per step once charging
+	var/speedperstep = 0.2
+	///MAXIMUMSPEED INCREASE
+	var/maxspeed = 2.4
+	///the amount by which the stam loss is multiplied by per step taken when charging
+	var/stamlossmult = 1
+	///the multiplier for damage inflicted when you bayonet charge someone
+	var/damagemult = 1.5
+	///This is a speed mult for when you finish charging
+	var/postspeedmult = 1.5
+	///the amount of time the post charge speed boost lasts (in deciseconds)
+	var/postspeedtime = 50
+	///the amount of time the stun lasts in 1/10ths of a second
+	var/slownesstime = 5
+	///the amount of time cooldown lasts in deciseconds
+	var/cooldowntime = 150
+
+	//Below are variables that are used internally. Do not externally change
+	var/stepstaken = 0
+	var/chargedir = null
+
+	var/amcharging = FALSE
+	var/chargewhenmove = FALSE
+	var/obj/item/weapon/gun/weaponinhand
+
+	///represents the time when the stepstaken var resets
+	var/movementcontinuationlimit
+	///the guy that is charging
+	var/mob/living/carbon/human/charger
+	///the child action of this component
+	var/datum/action/bayonetcharge/child
+
+/datum/component/bayonetcharge/Initialize(...)
+	. = ..()
+	updatevalues()
+
+/datum/component/bayonetcharge/proc/updatevalues()
+	weaponinhand = src.parent
+	charger = weaponinhand.gun_user
+	child = new
+	child.give_action(src.charger)
+	child.parent = src
+
+
+
+/datum/action/bayonetcharge
+	name = "Toggle Bayonet Charge"
+	action_icon_state = "charge_human"
+
+	var/datum/component/bayonetcharge/parent
+
+/datum/action/bayonetcharge/give_action(mob/M)
+	. = ..()
+
+/datum/action/bayonetcharge/remove_action(mob/M)
+
+
+	parent.charge_off()
+	. = ..()
```

Comment:
what is going on here

Reply:
It resets the signals and removes the selected frame from the action icon. It also procs stop_charge. stop_charge removes the steptaken var and removes the movement speed

Reply:
return ..()